packages/components/src/components/toast/toast.tsx

```diff
@@ -0,0 +1,86 @@
+import { Component, Prop, Method, h } from "@stencil/core";
+import { CssClassMap } from "../../utils/utils";
+import classNames from "classnames";
+import { formatDistance, subSeconds } from "date-fns";
+
+@Component({
+  tag: "t-toast",
+  styleUrls: ["toast.css"],
+  shadow: true
+})
+export class Toast {
+  /** (required) Alert class */
+  @Prop() customClass?: string = "";
+  @Prop() size?: string = "";
+  @Prop() theme?: string = "";
+  @Prop() variant?: string = "";
+  @Prop({ reflectToAttr: true }) opened?: boolean;
+  @Prop() autohide?: boolean = true;
+  @Prop() animated?: boolean = true;
+  /** (optional) Toast time */
+  @Prop() time?: number;
+
+  private getCssClassMap(): CssClassMap {
+    return classNames(
+      "toast",
+      this.customClass && this.customClass,
+      this.size && `toast--size-${this.size}`,
+      this.theme && `toast--theme-${this.theme}`,
+      this.variant && `toast--variant-${this.variant}`
+    );
+  }
+
+  private autohideTime = 5000;
+
+  onCloseToast = () => {
+    this.opened = false;
+    console.log("onCloseToast");
+  };
+
+  @Method()
+  async openToast() {
+    this.opened = true;
+  }
+
+  getTime = () => {
+    const formattedTime =
+      this.time &&
+      formatDistance(subSeconds(this.time, 3), new Date(), { addSuffix: true });
+    console.log(formattedTime);
+    return formattedTime;
+  };
+
+  onHideToast = () => {
+    if (this.autohide !== false) {
+      setTimeout(this.onCloseToast, this.autohideTime);
```

Comment:
You need to make sure that the `toast` only can be opened once, and that the timeout is also only fired once. Currently if you call `openToast()` multiple times it creates multiple timeouts which leads to buggy behaviour. Also you'll need to `clearTimeout()` once it's finished or aborted (component not on page anymore e.g. page switch) see https://www.w3schools.com/jsref/met_win_cleartimeout.asp for more details. 

You can debug the timeout doing something similar to this: 
```javascript
if (this.opened && this.autohide !== false) {
    const myTimeout = setTimeout(this.onCloseToast, this.autohideTime);
    console.log('myTimeout', myTimeout)
    return myTimeout;
} else {
    return null;
}
```

Reply:
![image](https://user-images.githubusercontent.com/27810942/67564232-d706b300-f722-11e9-8978-34fdcc106e7e.png)
the timeout is still firing multiple times..